### ISSUE TITLE
Activating a directory that doesn't exist

### ISSUE THREAD

**Issue by EricForgy** [JuliaLang/Pkg.jl#1654]
👋 

I'm on a Windows 10 machine running Julia v1.3.1 in Ubuntu WSL2.

I have a setup that looks like:

```
~/dev/MyApplication
|- docker-compose.yml
|- MyPackageA
   |- Dockerfile
   |- Project.toml
   |- src
      |- MyPackageA.jl
|- MyPackageB
   |- Dockerfile
   |- Project.toml
   |- src
      |- MyPackageB.jl
```

Both `MyPackageA.jl` and `MyPackageB.jl` are private packages and `MyPackageB.jl` depends on `MyPackageA.jl`, but at this point `MyPackageB / Project.toml` is still empty.

I activate `MyPackageB.jl` in order to add `MyPackageA` as a dependency and from within the directory `~/MyApplication`, I try:

```julia
(MyPackageB) pkg> dev ./MyPackageA
```

resulting in the following:

```julia
(MyPackageB) pkg> dev ./MyPackageA
 Resolving package versions...
  Updating `~/dev/MyApplication/~/dev/MyApplication/MyPackageB/Project.toml`
  [********] + MyPackageA [`../../../../../MyPackageA`]
  Updating `~/dev/MyApplication/~/dev/MyApplication/MyPackageB/Manifest.toml`
```
and it creates a new directory `'~'` under `~/dev/MyApplication` with a newly created - not updated - `Project.toml` and a new `Manifest.toml`.

I expected this to update `~/dev/MyApplication/MyPackageB/Project.toml` not create a new one in a weird new directory.

I used this approached effectively in Windows, where it works fine, but doesn't seem to work on Linux. It obviously has something to do with paths getting mangled somehow.

Any ideas?

**Comment by EricForgy** [comment]
I found the issue. 

When I activated `PackageB`, I did it by putting the following in my `startup.jl` (an artifact from the fact that I launch Julia in Windows by clicking the icon and not from the command line):

```julia
using Pkg; Pkg.activate("~/MyApplication/MyPackageB")
```

and didn't notice that it said:

```julia
Activating new environment at `~/dev/MyApplication/~/dev/MyApplication/MyPackageB/Project.toml`
```

That is weird. I'd expect it to error if it tries to activate some location that doesn't exist. At this point it doesn't error and it doesn't create a `Project.toml` / `Manifest.toml` in a newly created directory (yet).

Only after I try `dev ./MyPackage` from `~/dev/MyApplication` does it create both a new directory

```julia
~/dev/MyApplication/MyPackageB/~/dev/MyApplication/MyPackageB`
```

with new `Project.toml` / `Manifest.toml` inside that new directory.

So it looks like that `activate` function has an issue with `~`. 

I'll leave this issue open even though I found a workaround because I think 

```julia
using Pkg; Pkg.activate("~/MyApplication/MyPackageB")
```

should work.

My workaround is to put the full path in `actiate`, i.e.

```julia
using Pkg; Pkg.activate("/home/eric/dev/MyApplication/MyPackageB")
```

This works.

**Comment by KristofferC** [comment]
This seems consistent. `~` is never expanded in function calls (both Base and Pkg functions)

**Comment by EricForgy** [comment]
I'm new to running Julia in Linux and if we're not to expect Julia to expand `~`, that is fine. I call it a learning experience. However, I'm not sure this issue should be closed because I think there is a real problem here. I can change the name of the issue or something, but Julia should probably error if you activate a directory that doesn't exist, right?

If not, fine 🤦‍♂

**Comment by KristofferC** [comment]
> but Julia should probably error if you activate a directory that doesn't exist, right?

Maybe. Or at least if the directory above the directory you try to activate doesn't exist.

**Comment by sylvaticus** [comment]
Yes, I agree.. I activated by error a wrong folder and all went (apparently) normal - at the beginning of course :-/
I think as minimum there should be a warning (if not a real error) if one try to activate a non-existing dir... the "new" in the message goes looked behind too easily...
